plugins/ibm_qradar/icon_ibm_qradar/actions/update_offense/action.py

```diff
@@ -0,0 +1,93 @@
+import insightconnect_plugin_runtime
+import requests
+from insightconnect_plugin_runtime.exceptions import ClientException, PluginException
+
+from .schema import UpdateOffenseInput, UpdateOffenseOutput, Component, Input, Output
+
+from icon_ibm_qradar.util.constants.endpoints import UPDATE_OFFENSES_ENDPOINT
+from icon_ibm_qradar.util.constants.messages import (
+    EMPTY_OFFENSE_ID_FOUND,
+    CLOSING_REASON_ID_NOT_PROVIDED,
+    CLOSING_REASON_ID_PROVIDED_FOR_OTHER_STATUS,
+    CLOSING_REASON_ID_PROVIDED_IS_NOT_INTEGER,
+)
+from icon_ibm_qradar.util.url import URL
+from icon_ibm_qradar.util.utils import prepare_request_params, handle_response
+
+
+class UpdateOffense(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super().__init__(
+            name="update_offense",
+            description=Component.DESCRIPTION,
+            input=UpdateOffenseInput(),
+            output=UpdateOffenseOutput(),
+        )
+        self.endpoint = UPDATE_OFFENSES_ENDPOINT
+
+    def run(self, params={}):
+        """Run Method to execute action.
+
+        :param params: Input Param config required for the Action
+        :return:
+        """
+        offense_id = params.get(Input.OFFENSE_ID, "")
+        self.logger.info("Offense ID provided: %s", offense_id)
+
+        assigned_to = params.get(Input.ASSIGNED_TO, "")
+        self.logger.info("assigned_to provided: %s", assigned_to)
+
+        closing_reason_id = params.get(Input.CLOSING_REASON_ID, "")
+        self.logger.info("closing_reason_id provided: %s", closing_reason_id)
+
+        if closing_reason_id != "" and not closing_reason_id.isdigit():
+            raise ClientException(Exception(CLOSING_REASON_ID_PROVIDED_IS_NOT_INTEGER))
+
+        follow_up = params.get(Input.FOLLOW_UP, False)
+        self.logger.info("follow_up provided: %s", follow_up)
+
+        protected = params.get(Input.PROTECTED, False)
+        self.logger.info("protected provided: %s", protected)
+
+        status = params.get(Input.STATUS, "")
+        self.logger.info("status provided: %s", status)
+
+        query_params = {}
+
+        if assigned_to != "":
+            query_params["assigned_to"] = assigned_to
+
+        if closing_reason_id != "":
+            if status != "CLOSED":
+                self.logger.info("Terminating: Closing Reason Id is provided to open / hide offense.")
+                raise ClientException(Exception(CLOSING_REASON_ID_PROVIDED_FOR_OTHER_STATUS))
```

Comment:
```suggestion
                raise ClientException(Exception(CLOSING_REASON_ID_PROVIDED_FOR_OTHER_STATUS))
```
Please raise a PluginException